test/E2ETests/TestAssets/DataRowTestProject/DataRowTests_DisplayNemeMethodAndDeclaringType.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Reflection;
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace DataRowAttributeTestProject;
+
+[TestCategory("GetCustomDisplayName")]
+[TestClass]
+public class DataRowTests_DisplayNemeMethodAndDeclaringType
+{
+    [TestMethod]
+    [DataRow(1,DisplayNameMethod = "GetCustomDisplayName")]
+    public void DataRowTest_WithDisplayNameMethod_passes(int number)
+    {
+        Assert.AreEqual(number,1);
+    }
+
+    [TestMethod]
+    [DataRow(1, DisplayNameMethod = "GetCustomDisplayNameOtherType", DisplayNameMethodDeclaringType =typeof(Dummy))]
+    public void DataRowTest_WithDisplayNameMethodOtherType_Passes(int number)
+    {
+        Assert.AreEqual(number, 1);
+    }
+
+    [TestMethod]
+    [DataRow(1, DisplayNameMethod = "GetCustomDisplayNamePrivate", DisplayNameMethodDeclaringType = typeof(Dummy))]
```

Comment:
This method is not defined on type `Dummy`. Might be worth to have 2 tests: one for bad "accessibility" and one for method not found on type XXX.